src/frontend/qt_sdl/main.cpp

```diff
@@ -1399,16 +1403,52 @@ QString MainWindow::loadErrorStr(int error)
     }
 }
 
+std::string extractROM(char* zipName, std::string zipDir){
+    //Open the ZIP archive
+    int err = 0;
+    zip *z = zip_open(zipName, 0, &err);
+    
+	struct zip_stat st;
+    zip_stat_init(&st);
+    zip_stat_index(z, 0, 0, &st); //Get information about the file at index 0
+	
+    //Allocate memory for its uncompressed contents
+    char *contents = new char[st.size];
```

Comment:
we make the star on the left side

contents needs to be deleted

prefer u8 over char for a binary stream

Reply:
Alright, i've made it delete contents.
Also, when I change it's type to u8, it gives me errors like `error: invalid conversion from 'u8*' {aka 'unsigned char*'} to 'const char*'`

Reply:
then cast it to char*

Reply:
ok, done it.